packages/components/inputs/selectable-search-input/src/selectable-search-input.tsx

```diff
@@ -0,0 +1,435 @@
+// TODO: @redesign cleanup
+import {
+  type MouseEvent,
+  type KeyboardEvent,
+  type ChangeEvent,
+  type ReactNode,
+  useState,
+  useCallback,
+  useRef,
+} from 'react';
+import SecondaryIconButton from '@commercetools-uikit/secondary-icon-button';
+import Constraints from '@commercetools-uikit/constraints';
+import { SearchIcon, CloseIcon } from '@commercetools-uikit/icons';
+import {
+  createSequentialId,
+  filterDataAttributes,
+  warning,
+} from '@commercetools-uikit/utils';
+import {
+  getClearIconButtonStyles,
+  getSearchIconButtonStyles,
+  getSelectableSearchInputContainerStyles,
+  getSelectableSearchInputStyles,
+} from './selectable-search-input.styles';
+import SelectableSelect from './selectable-select';
+import { useFieldId, useToggleState } from '@commercetools-uikit/hooks';
+import styled from '@emotion/styled';
+import { designTokens, useTheme } from '@commercetools-uikit/design-system';
+import { css } from '@emotion/react';
+import { type Props as ReactSelectProps } from 'react-select';
+
+const Container = styled.div`
+  display: flex;
+`;
+
+const getTextInputName = (name?: string) =>
+  name ? `${name}.textInput` : undefined;
+
+const getDropdownName = (name?: string) =>
+  name ? `${name}.dropdown` : undefined;
+
+type TCustomEvent = {
+  target: {
+    id?: string;
+    name?: string;
+    value?: string | string[] | null;
+  };
+  persist?: () => void;
+};
+
+export type TValue = {
+  textValue: string;
+  dropdownValue: string;
+};
+
+export type TOption = {
+  value: string;
+  label?: ReactNode;
+};
+
+export type TOptionObject = {
+  options: TOption[];
+};
+
+export type TSelectableSearchInputProps = {
+  /**
+   * Used as HTML id property. An id is auto-generated when it is not specified.
+   */
+  id?: string;
+  /**
+   * Used as HTML autocomplete property
+   */
+  autoComplete?: string;
+  /**
+   * Indicate if the value entered in the input is invalid.
+   */
+  'aria-invalid'?: boolean;
+  /**
+   * HTML ID of an element containing an error message related to the input.
+   */
+  'aria-errormessage'?: string;
+  /**
+   * Used as HTML name of the input component property.
+   */
+  name?: string;
+  /**
+   * Value of the input. Consists of text input and selected option.
+   */
+  value: TValue;
+  /**
+   * Called with the event of the input or dropdown when either the selectable dropdown or the text input have changed.
+   */
+  onChange?: (event: TCustomEvent) => void;
+  /**
+   * Called when input is blurred
+   */
+  onBlur?: (event: TCustomEvent) => void;
+  /**
+   * Called when input is focused
+   */
+  onFocus?: (event: TCustomEvent) => void;
+  /**
+   * Handler when the search button is clicked.
+   */
+  onSubmit: (searchValue: string) => void;
```

Comment:
I think the `onSubmit` handler should be called with both the searchValue and the selected option from the dropdown so the consumers can know what the user selected to trigger the right query.

Reply:
I was thinking about this also. I decided the handler should be called with on the seachTerm because the component on MC does the same but, It makes more sense for the consumer to have both values. Thanks.